misc/json-to-record-converter/README.md

```diff
@@ -0,0 +1,530 @@
+                        <li>JsonObject</li>
+                    </ul>
+                </td>
+                <td>Object</td>
+                <td>record</td>
+            </tr>
+            <tr>
+                <td>
+                    <ul>
+                        <li>JsonArray</li>
+                    </ul>
+                </td>
+                <td>Array</td>
+                <td>array</td>
+            </tr>
+        </tbody>
+    </table>
+    <p>&nbsp;</p>
+</div>
+
+### Mapping of Null and Optional Types
+This section explains how different ways an optional Ballerina field can be generated.
+
+#### Scenario 01: JSON null
+When the value of a JSON field is null, it can be mapped as a field of [anydata](https://ballerina.io/spec/lang/master/#anydata) data type in ballerina.
+
+JSON
+```json
+{
+   "nullProperty": null
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   anydata nullProperty;
+};
+```
+
+#### Scenario 02: Optional fields within the same JSON object
+When there are two or more JSON Objects with the same name, and if there are any differencing fields those fields will also be considered as optional fields.
+
+JSON
+```json
+{
+  "object": {
+    "fieldOne": "StringValue",
+    "fieldTwo": 16
+  },
+  "object": {
+    "fieldOne": "StringValue"
+  }
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   string fieldOne;
+   int fieldTwo?;
+};
+```
+
+In the above example, both JSON object fields are with the same name. And there is an intersecting field which is **_fieldOne_** and the differencing field **_fieldTwo_**. Thus, the intersecting field is made to be required and the differencing field is made to be optional.
+
+### Mapping of Primitive Types
+This section explains how JSON primitives can be mapped to Ballerina records. The JSON primitive types are String, Boolean and Number. Any JSON primitive type will be parsed as a **_JsonPrimitive_** Java object. And each object can be checked for String, Boolean or Number by calling the relevant method on the object.
+
+#### Scenario  03: JSON string
+JSON string can be mapped to ballerina string type.
+
+JSON
+```json
+{
+  "stringProperty": "StringValue"
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   string stringProperty;
+};
+```
+
+#### Scenario  04: JSON boolean
+JSON boolean can be mapped to ballerina boolean
+
+JSON
+```json
+{
+  "booleanProperty": true
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   boolean booleanProperty;
+};
+```
+
+#### Scenario  05: JSON number
+JSON Numbers can be either of below,
+```json
+{
+  "positiveInteger": 210,
+  "negativeInteger": -210,
+  "floatingPointNumber": 21.05,
+  "exponentialNumber": 2.105E+1
+}
+```
+So, we have to map JSON numbers to either of Ballerina int, decimal or float based on the value. And, here we use all floating point numbers as decimals since, decimal is more appropriate to use since, decimals are more precise than floats and can store 128 bits
```

Comment:
```suggestion
JSON numbers should have to be mapped to either of the Ballerina int, decimal, or float types based on the value. Here, all floating point numbers are used as decimals since decimal is more appropriate to use and also because decimals are more precise than floats and can store 128 bits.
```
We do not use the word `we` in technical documentation. It should always be addressing the user directly saying `you`. If not, using passive voice is the best practice.